flink-streaming-connectors/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisConsumerTest.java

```diff
@@ -145,6 +145,39 @@ public void testUnrecognizableStreamInitPositionTypeInConfig() {
 	}
 
 	@Test
+	public void testStreamInitPositionTypeSetToAtTimestampButNoInitTimestampSetInConfig() {
+		exception.expect(IllegalArgumentException.class);
+		exception.expectMessage("Please set value for initial timestamp ('"
+			+ ConsumerConfigConstants.STREAM_INITIAL_TIMESTAMP + "') when using AT_TIMESTAMP initial position.");
+
+		Properties testConfig = new Properties();
+		testConfig.setProperty(ConsumerConfigConstants.AWS_REGION, "us-east-1");
+		testConfig.setProperty(ConsumerConfigConstants.AWS_CREDENTIALS_PROVIDER, "BASIC");
+		testConfig.setProperty(ConsumerConfigConstants.AWS_ACCESS_KEY_ID, "accessKeyId");
+		testConfig.setProperty(ConsumerConfigConstants.AWS_SECRET_ACCESS_KEY, "secretKey");
+		testConfig.setProperty(ConsumerConfigConstants.STREAM_INITIAL_POSITION, "AT_TIMESTAMP");
+
+		KinesisConfigUtil.validateConsumerConfiguration(testConfig);
+	}
+
+	@Test
+	public void testUnparsableDateForInitialTimestampInConfig() {
+		exception.expect(IllegalArgumentException.class);
+		exception.expectMessage("Invalid value given for initial timestamp for AT_TIMESTAMP initial position in stream. "
```

Comment:
I would suggest to not let the expect message be this verbose in tests.
This increases the likeliness that the tests will need to be altered, whenever we want to tweak the messages a bit.
I think `Invalid value given for initial timestamp for AT_TIMESTAMP initial position in stream` is enough.